unifi/device.go

```diff
@@ -0,0 +1,41 @@
+package unifi
+
+import (
+	"context"
+)
+
+func (c *Client) ListDevice(ctx context.Context, site string) ([]Device, error) {
+	return c.listDevice(ctx, site)
+}
+
+func (c *Client) GetDevice(ctx context.Context, site, mac string) (*Device, error) {
+	return c.getDevice(ctx, site, mac)
+}
+
+func (c *Client) DeleteDevice(ctx context.Context, site, id string) error {
+	return c.deleteDevice(ctx, site, id)
+}
+
+func (c *Client) CreateDevice(ctx context.Context, site string, d *Device) (*Device, error) {
+	return c.createDevice(ctx, site, d)
+}
+
+func (c *Client) UpdateDevice(ctx context.Context, site string, d *Device) (*Device, error) {
+	return c.updateDevice(ctx, site, d)
+}
+
+func (c *Client) GetDeviceById(ctx context.Context, site, id string) (*Device, error) {
```

Comment:
does the generator `getDevice` not work? not sure how devices work in this case

also can just call this `GetDevice`

```suggestion
func (c *Client) GetDevice(ctx context.Context, site, id string) (*Device, error) {
```

Reply:
If there is a reason you can't just use `getDevice` I think just comment in here so I don't accidentally change it back in the future 😂

Reply:
So, the weird thing about the Unifi REST API is you can only get a single device using the device's MAC address as the key. It fails when you use the ID, unlike everything else in the API.

Options:
* Leave the default as is and make it clear that it's using the MAC address when getting a single device
* Swap names of the functions for consistency with the rest of the API (despite the fact that `GetDeviceById` actually pulls all items and iterates through a list, making it more expensive a call)
* Change all of the functions to be explicit, leaving no "default" `GetDevice` for anyone to call

I may go the route of being explicit to make it completely clear to the end user.

Reply:
I think if its going to be ID, unless we are going to change all the `Get*` methods in the package, it should just be `GetDevice` even if its iterating through a list (can always mention that in the doc comment. But makes sense to add a `GetDeviceByMAC`, there already is a precedent with `GetUser` and `GetUserByMAC`